tasks/create-pyxis-image/create-pyxis-image.yaml

```diff
@@ -110,24 +110,37 @@ spec:
         echo "${pyxisKey}" > /tmp/key
 
         COMPONENTS=$(jq '.components | length' "${SNAPSHOT_SPEC_FILE}")
+        JSON_OUTPUT='{}'
         for (( i=0; i < $COMPONENTS; i++ )); do
             CONTAINER_IMAGE=$(jq -r ".components[${i}].containerImage" "${SNAPSHOT_SPEC_FILE}")
+            JSON_OUTPUT=$(jq ".components[$i] += { \"containerImage\": \"${CONTAINER_IMAGE}\", \"pyxisImages\": []}" \
```

Comment:
I would suggest using jq args here - that way you can avoid all the escaping (`\"`).

Like: `jq --argjson id $i --arg image ${CONTAINER_IMAGE} '.components[$id] += {"containerImage": "$image", ...}'`

(The reason for the first `--argjson` instead of `--arg` is because otherwise it would be a string and we want it as integer.)

Reply:
true. Totally forgot about those `args`  parameters. changed.